CIP-0164/README.md

```diff
@@ -0,0 +1,2774 @@
+| ------------------------------------------------------ | ---------------------------------------------------------------- |
+| ![Mean nodal ingress](images/ingress-average-area.svg) | ![Mean CPU load among all nodes](images/cpu-mean-timeseries.svg) |
+
+<em>Figure 15: Mean nodal ingress (left) and Mean CPU load among all nodes
+(right)</em>
+
+</div>
+
+<div align="center">
+<a name="figure-16" id="figure-16"></a>
+
+![Mean CPU load among all nodes](images/cpu-mean-histogram.svg)
+
+<em>Figure 16: Mean CPU load among all nodes ("Gen" = generated, "Val" =
+validated, "RH" = ranking block header, "RB" = ranking block body, "EH" =
+endorser block header, "EB" = endorser block body", "TX" = transaction)</em>
+
+</div>
+
+Note that the transaction workload in the simulations above was modeled upon the
+_average_ amount of Plutus computation typical of the Cardano mainnet. The low
+time-averaged CPU usage in the simulations (i.e., less than 15% of a vCPU)
+suggests that the per-transaction and/or per-block Plutus budget could be
+significantly increased under Leios: either every transaction could have a
+modestly higher budget, or some transactions could use an order of magnitude
+more Plutus execution units. Statistical analysis of [CPU usage in ledger
+operations][timings] using the [db-analyser tool][dbanalyser] on Cardano mainnet
+from epoch 350 through 573 yields the following simple models of the CPU cost of
+validating signatures and executing Plutus in the transactions of a block.
+Because of the noisiness in the raw mainnet data, these estimates are uncertain.
+
+- Ledger "apply" operation, consisting of phase 1 & 2 validation along with
+  updating the current ledger state:
+  - CPU per transaction in a block: `620.1 μs/tx`.
+  - Linear model that accounts for Plutus:
+    `(262.4 μs/tx) * (number of transactions) + (948.7 μs/Gstep) * (billions of Plutus execution steps)`.
+- Ledger "reapply" operation, consisting of updating the current ledger state,
+  omitting previously performed phase 1 & 2 validation:
+  - Linear model: `(353.9 μs) + (21.51 μs/kB) * (size of the block)`
+  - Linear model that accounts for Plutus:
+    `(347.8 μs) + (19.43 μs/kB) * (size of the block) + (21.27 μs/Gstep) * (billions of Plutus execution steps)`
+
+The Leios simulators perform the "apply" operation when a transaction is first
+seen, either when it is received for the memory pool or when it is fetched after
+first being seen in an RB or EB; they perform the "reapply" operation when a
+block is being generated or validated. A more nuanced model of CPU usage in the
+simulators would account for Plutus execution explicitly, but the linear models
+described above are used to account for Plutus workloads implicitly. The
+following plot of simulation results limit each node to 4 vCPU cores and suggest
+that workloads of 20,000e9 Plutus execution steps per EB may be feasible: this
+is 1000 times the current Cardano mainnet limit of 20e9 steps for Praos blocks.
```

Comment:
```suggestion
that workloads of 2e13 Plutus execution steps per EB may be feasible: this
is 1000 times the current Cardano mainnet limit of 2e10 steps for Praos blocks.
```
Perhaps use the `e` notation like this, or something more explicit like $20\cdot 10^{12}$.